### DISCUSSION TITLE
Assets not compiled when creating a "web" application

### DISCUSSION THREAD

**Discussion by simonjcarr** [adonisjs/core#4208] [Help] [answered]
When I create a new Adonis 5 application using the "web" option, assets are not compiled. 
I am using Node 18.17.0 on Ubuntu 22.04 Desktop
I install adonis using `npm init adonis-ts-app .`

Here is a repo with my code. [git@github.com:simonjcarr/adonis_webpack_not_working.git](https://github.com/simonjcarr/adonis_webpack_not_working.git)

After installation I run `npm run dev`, the application runs and is hosted on http://localhost:3333

Here is what is shown in the browser

![image](https://github.com/adonisjs/core/assets/1474382/47d5590e-ee95-4c43-8c60-23eec62913c2)

This is what is in my assets folder
![image](https://github.com/adonisjs/core/assets/1474382/6bfbab05-e132-4510-b997-4b54d5eaade1)

No changes have been made to the code from the boilerplate created by the install.

Can anyone advise if I am doing something wrong?

**Comment by WeyIin** [comment]
I cloned your repo and everything seem to be working? 

![image](https://github.com/adonisjs/core/assets/5105137/dee882c4-ff67-4c4e-be64-27013ada31a1)

**Reply by simonjcarr** [reply]
@WeyIin Thanks for the quick response. Can you confirm your version of Node, NPM and what OS your running on?

I have cloned from my repository, ran npm install, then npm run dev. I get the same problem.

When I start the app, this is the output I get

```
npm run dev

> manageix@1.0.0 dev /home/simon@soxprox.lan/dev/adonis_webpack_not_working
> node ace serve --watch

[ info ]  building project...
[ info ]  starting http server...
[07:55:40.580] INFO (manageix/3748968): started server on 0.0.0.0:3333
[ encore ] Running webpack-dev-server ...
[ info ]  watching file system for changes
╭────────────────────────────────────────────────────────╮
│                                                        │
│    Server address: http://127.0.0.1:3333               │
│    Watching filesystem for changes: YES                │
│    Encore server address: http://localhost:8080        │
│                                                        │
╰────────────────────────────────────────────────────────╯
UPDATE: public/assets/manifest.json
UPDATE: public/assets/entrypoints.json
[ encore ]  DONE  Compiled successfully in 526ms07:55:41
[ encore ] webpack compiled successfully

```

As you can see public/assets/manifest.json and entrypoints.json are created, but it does not create the css or js files.

**Comment by WeyIin** [comment]
restart the server, run `npm run dev` again, if nothing works reinstall npm packages. Delete `node_modules` and do `npm install`

**Comment by simonjcarr** [comment, marked as answer]
I solved the problem. Port 8080 was already in use, I think by docker. The funny thing is that when I ran netstat -tulpn port 8080 was not listed as in use, but even without Adonis running, I was getting a 404 when I went to http://localhost:8080. In the end I stopped a few services one of them being docker, and suddenly everything worked fine.

Thanks for your help anyway.
